packages/medusa/src/services/discount-condition.ts

```diff
@@ -0,0 +1,237 @@
+import { MedusaError } from "medusa-core-utils"
+import { BaseService } from "medusa-interfaces"
+import { EntityManager } from "typeorm"
+import { EventBusService } from "."
+import { Discount, DiscountCondition, DiscountConditionType } from "../models"
+import { DiscountRepository } from "../repositories/discount"
+import { DiscountConditionRepository } from "../repositories/discount-condition"
+import { DiscountRuleRepository } from "../repositories/discount-rule"
+import { UpsertDiscountConditionInput } from "../types/discount"
+import { PostgresError } from "../utils/exception-formatter"
+
+/**
+ * Provides layer to manipulate discount conditions.
+ * @implements {BaseService}
+ */
+class DiscountConditionService extends BaseService {
+  protected readonly manager_: EntityManager
+  protected readonly discountRuleRepository_: typeof DiscountRuleRepository
+  protected readonly discountRepository_: typeof DiscountRepository
+  protected readonly discountConditionRepository_: typeof DiscountConditionRepository
+  protected readonly eventBus_: EventBusService
+
+  constructor({
+    manager,
+    discountRuleRepository,
+    discountRepository,
+    discountConditionRepository,
+    eventBusService,
+  }) {
+    super()
+
+    /** @private @const {EntityManager} */
+    this.manager_ = manager
+
+    /** @private @const {DiscountRuleRepository} */
+    this.discountRuleRepository_ = discountRuleRepository
+
+    /** @private @const {DiscountRepository} */
+    this.discountRepository_ = discountRepository
```

Comment:
thought: I strongly think that we should aim toward allowing only 1 service to be dependent on any repository. I.e. since the DiscountService is dependent on the `discountRuleRepo + discountRepo` I think we should avoid it here. The reasoning is to avoid coupling services and to ensure that we don't manipulate the data store from multiple angles which is likely going to cause us trouble down the line.

Reply:
Since the condition can't exists without a discount I think they are then part of the same domain. That said, I think that we can remove the repository from here and just inject the discount services since they are coupled by nature, because as I said the condition need a discount to exists wdyt?

Reply:
I don't think we can inject the discount service since it will introduce a circular dependency and awilix breaks. It would be ideal since it would also clean up the `upsertCondition` method.